ARSIS-Telemetry-GroundControl/telemetry-api/app/routers/uia.py

```diff
@@ -0,0 +1,46 @@
+from fastapi import APIRouter
+from pydantic import BaseModel
+from app.database import connection
+
+router = APIRouter(prefix="/uia", tags=["uia"])
+
+class UIAStatusRequest(BaseModel):
+    o2: bool | None = None
+    power: bool | None = None
+    comm: bool | None = None
+
+@router.get("/")
+async def get_uia():
+    with connection.cursor() as db:
+        db.execute("SELECT * FROM uia WHERE id = 1;")
+        (id, o2, power, comm, updatedAt) = db.fetchone()
+        response = {
+            "o2": o2,
+            "power": power,
+            "comm": comm,
+            "updatedAt": updatedAt
+        }
+        return response
+
+@router.post("/")
+async def post_uia(body: UIAStatusRequest):
+    with connection.cursor() as db:
+        if body.o2 != None:
+            db.execute("UPDATE uia SET o2 = %s, updatedAt = now() WHERE id = 1;", (body.o2,))
+
+        if body.power != None:
+            db.execute("UPDATE uia SET power_ = %s, updatedAt = now() WHERE id = 1;", (body.power,))
+
+        if body.comm != None:
+            db.execute("UPDATE uia SET comm = %s, updatedAt = now() WHERE id = 1;", (body.comm,))
```

Comment:
Can we consolidate this to be one transaction?

Reply:
Also we should use f strings they are faster (last I checked) and more ready

Reply:
ive tried but its kind of hard to create a single statement if we wanna have optional fields (i couldnt find a better solution 😓) as the statement will try to parse None types into null. im using psycopg2 to handle the statement parsing as i dont wanna handle when they provide unexpected values in the request body (preventing sql injection).

Reply:
actually i should do some validation beforehand and then do the comprehension incase the values arent in the same expected type and return status code 400 if they arent

Reply:
I'm pretty sure the validation is done internally by fastAPI. if it receives a body that isnt in the form of UIAStatusRequest it will either try to cast it or will reject it

Reply:
I think its an unprocessable entity 422? but I would have to check